### PR TITLE
Add support for Braket Devices enums and S3 buckets

### PR DESCRIPTION
**Pull request by maliasadi**
**Context:**
This PR fixes https://github.com/PennyLaneAI/catalyst/issues/206 by adding support for 
- [Braket Devices enums](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/src/braket/devices/devices.py)
-  user-provided `s3_destination_folder` in remote calls.

[sc-45094]

### CODE REVIEW

File: frontend/test/pytest/test_braket_remote_devices.py [hunk 1]
```diff
@@ -788,8 +838,8 @@ def f(x):
         @qjit
         def compiled_grad_default(x: float):
             g = qml.qnode(device)(f)
-            h = grad(g, h=1e-4)
-            i = grad(h, h=1e-4)
+            h = grad(g, method="fd", h=1e-4)
+            i = grad(h, method="fd", h=1e-4)
             return i(x)
```
Comment (dime10) [inline frontend/test/pytest/test_braket_remote_devices.py:843]: Oops good catch 👍

**Review by dime10 — APPROVED** [review]
Thanks @maliasadi 🎉

**Review by josh146 — APPROVED** [review]
Nice, thanks @maliasadi!

I have a separate question: your ADR on allowing devices to specifying compilation options. Would it be sufficient in allowing this logic here to not be hardcoded into the Catalyst functions?

**Comment by maliasadi** [comment]
> I have a separate question: your ADR on allowing devices to specifying compilation options. Would it be sufficient in allowing this logic here to not be hardcoded into the Catalyst functions?

@josh146 the `qml.device` method can take any arbitrary keyword arguments and, we may need to embed these kwargs to the generated LLVM IR which will be parsed and used at run time by the runtime. We don't really need to hardcode these kwargs in the frontend if we can capture these kwargs or the device itself communicates these arguments. However, this issue is unique to the Braket devices at the moment and one possible solution is by simply adding a method that returns a dictionaries of device specifications. For the Braket plugin, this's `device_arn` and `s3_destination_folder`.

**Comment by josh146** [comment]
Thanks @maliasadi!

> However, this issue is unique to the Braket devices at the moment and one possible solution is by simply adding a method that returns a dictionaries of device specifications. For the Braket plugin, this's device_arn and s3_destination_folder.

Over time we will run into this issue with other devices as well. Even lightning has additional options you can pass it -- e.g., for MCMC sampling:

```python
dev = qml.device("lightning.qubit", wires=2, shots=1000, mcmc=True, kernel_name="NonZeroRandom", num_burnin=200)
```

**Pull request merged**